defender/downloaded/Compound_Governance_Automation.js

```diff
@@ -0,0 +1,143 @@
+  const governanceContract = new ethers.Contract(
+    compoundGovernanceAddress,
+    compoundGovernanceAbi,
+    signer,
+  );
+
+  // create the client to interact with the key-value store
+  console.debug('Creating KeyValueStoreClient');
+  const store = new KeyValueStoreClient(autotaskEvent);
+
+  // get the value from the key-value store
+  // expect a comma separated list of integers
+  console.debug('Retrieving proposalIdsToIgnore');
+  let proposalIdsToIgnore = await store.get('proposalIdsToIgnore');
+  console.debug(`proposalIdsToIgnore retrieved: ${proposalIdsToIgnore}`);
+
+  // is this the first time the Autotask has been executed
+  if (proposalIdsToIgnore === undefined || proposalIdsToIgnore === null) {
+    proposalIdsToIgnore = [];
+  } else if (proposalIdsToIgnore.length === 0) {
+    // handle the empty string case
+    proposalIdsToIgnore = [];
+  } else {
+    // non-empty string case
+    proposalIdsToIgnore = proposalIdsToIgnore.split(',').map((value) => parseInt(value, 10));
+  }
+  console.debug(`proposalIdsToIgnore formatted: ${proposalIdsToIgnore}`);
+
+  // check the initialized value of the proposal ID
+  // NOTE: the first proposal ID is this value PLUS 1
+  const initialProposalId = (await governanceContract.initialProposalId()).toNumber();
+  const startProposalId = initialProposalId + 1;
+  console.debug(`first proposal value: ${startProposalId}`);
+
+  // check the value of the latest proposal ID
+  const currentProposalId = (await governanceContract.proposalCount()).toNumber();
+  console.debug(`currentProposalId: ${currentProposalId}`);
+
+  // create the Array of proposal IDs to check
+  const proposalsToCheck = [];
+  for (let proposalId = startProposalId; proposalId <= currentProposalId; proposalId++) {
+    if (proposalIdsToIgnore.includes(proposalId) === false) {
+      proposalsToCheck.push(proposalId);
+    }
+  }
+  console.debug(`proposalsToCheck: ${proposalsToCheck}`);
+
+  const promises = proposalsToCheck.map(async (proposalId) => {
+    const state = await governanceContract.state(proposalId);
+    let callExecute;
+    switch (state) {
+      case 0: // Pending
+      case 1: // Active
+        // nothing to do
+        break;
+      case 2: // Canceled
+      case 3: // Defeated
+      case 6: // Expired
+      case 7: // Executed
+        // add to ignore Array for next time
+        console.debug(`Adding proposal ID ${proposalId} to proposalIdsToIgnore`);
+        proposalIdsToIgnore.push(proposalId);
+        break;
+      case 4: // Successful
+        // execute transaction with Relay to call queue()
+        console.debug(`Calling queue for proposal ID ${proposalId}`);
+        await governanceContract.queue(ethers.BigNumber.from(proposalId));
+        // intentionally allow fall-through so that a successfully queued proposal has the
+        // opportunity to be executed as soon as possible
+        // eslint-disable-next-line no-fallthrough
+        /*
+      case 5: // Queued
```

Comment:
I see that you removed execution, should we also comment out the `successful -> queue` step as well?

Reply:
removed